### PR TITLE
Create a basic persistence layer

### LINKED ISSUE #70: Create a simple persistence layer
The backend will use the python framework Flask with SQLAlchemy to implement exporting and importing documents to and from a server.

### PR DESCRIPTION
**Pull request by JeHugawa**
Closes #70

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class swagger(db.Model):
+    id = db.Column(db.Integer, primary_key = True)
+    swagger = db.Column(db.LargeBinary(), unique=True)
+    
+    def __init__(self, swagger):
+        self.swagger = swagger
+
+    def __repr__(self):
+        return '<swagger %r>' % self.swagger
+
```
Comment (mauriciovieira) [inline backend/main.py]: Remove extra blank lines. 
Actually, my suggestion is to run flake8

File: backend/init.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from main import db
```
Comment (mauriciovieira) [inline backend/init.py]: Run flake8.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
 node_modules
 .idea
 .DS_STORE
+*.pyc
+main.db
```
Comment (mauriciovieira) [inline .gitignore]: Please, write some documentation on how to setup this backend project.

Comment (JeHugawa) [inline .gitignore]: There is allready some comments in the code about setting up but i will write a separate readme for details later.

Comment (mauriciovieira) [inline .gitignore]: Thanks! It can be on README.md, actually. :-) Our current README here is so short ☹️

File: backend/init.py (hunk 2 above)
Comment (JeHugawa) [inline backend/init.py]: What is wrong with that line?

Comment (mauriciovieira) [inline backend/init.py]: nothing. but [python style guide](https://www.python.org/dev/peps/pep-0008/) says to add 2 blank spaces between imports and the code.

you can run [flake8](http://flake8.pycqa.org/en/latest/) and/or [pep8](https://pypi.python.org/pypi/pep8) to see the suggestions/corrections and make the code more pythonic.

Comment (tulir) [inline backend/init.py]: flake8, pycodestyle and yapf all leave the two lines as-is.

**Comment by mauriciovieira** [comment]
@JeHugawa you should rebase from master to fix `.gitignore` conflicts.

**Comment by mauriciovieira** [comment]
I strongly suggest the use of [virtualenv](http://python-guide-pt-br.readthedocs.io/en/latest/dev/virtualenvs/) and `requirements.txt` file. It's a common practice on python software projects, whereas suggesting to install a pip package globally is not advised.

**Comment by JeHugawa** [comment]
I would prefer not using virtuanlenv to be honest. But if someone really wants to use it, i think i can start using it.

File: backend/main.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        return '<Swagger %r>' % self.swagger
+
+
+@app.route('/', methods=['POST'])
+def upload():
+    temp = request.files['file']
+    swagger = temp.read()
+    db.session.add(Swagger(swagger))
+    db.session.commit()
+    return 'Added swagger to the database'
```
Comment (tulir) [inline backend/main.py]: This should probably return the ID of the new stored document.

File: backend/main.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from flask import *
+from werkzeug import secure_filename
```
Comment (tulir) [inline backend/main.py]: Remove unused imports

File: backend/main.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+@app.route('/', methods=['POST'])
+def upload():
+    temp = request.files['file']
+    swagger = temp.read()
+    db.session.add(Swagger(swagger))
+    db.session.commit()
+    return 'Added swagger to the database'
+
+
+app.run(debug = True)
```
Comment (tulir) [inline backend/main.py]: Shouldn't this be wrapped in
```python
if __name__ == "__main__":
```

**Comment by xylix** [comment]
Closing this because we created a (backend repository)[https://github.com/apinf/openapi-hub]. The python from here might be used there at some point, so let's not delete the branch yet.

**Pull request closed without merging**